### PR TITLE
se: simplify rendering of introspection warnings

### PR DESCRIPTION
**Pull request by tomhoule**
For the messages with groups. This achieves the same behaviour but with much less code and no new traits.

### CODE REVIEW

**Comment by tomhoule** [comment]
(not ready for review, I just had to check that my intuitions on yesterday's review were correct)

**Comment by tomhoule** [comment]
> (not ready for review, I just had to check that my intuitions on yesterday's review on #3924 were correct)

now ready

File: schema-engine/connectors/schema-connector/src/warnings.rs [hunk 1]
```diff
@@ -249,19 +174,16 @@ impl fmt::Display for Warnings {
 
         fn render_warnings_grouped<'a, T>(msg: &str, items: &'a Vec<T>, f: &mut fmt::Formatter<'_>) -> fmt::Result
         where
-            T: fmt::Display,
-            GroupBy<'a, T>: Groupable<String, T> + fmt::Display,
+            GroupBy<'a, T>: fmt::Display,
```
Comment (Druue) [inline schema-engine/connectors/schema-connector/src/warnings.rs:177]: 🎉

File: schema-engine/connectors/schema-connector/src/warnings.rs [hunk 2]
```diff
@@ -1,130 +1,55 @@
...
-    Ok(())
 }
 
-impl fmt::Display for GroupBy<'_, TypeAndField> {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        let grouped = self.group_by();
-        for (key, value) in grouped {
-            write!(f, "  - ")?;
-            fmt_group_type_and_field(f, &key, &value)?;
+fn display_list<T: Ord>(
```
Comment (Druue) [inline schema-engine/connectors/schema-connector/src/warnings.rs:27]: 🎉

**Review by Druue — APPROVED** [review]

File: schema-engine/connectors/schema-connector/src/warnings.rs [hunk 3]
```diff
@@ -1,130 +1,55 @@
...
+            key = Some(project_key(next));
         }
 
-        Ok(())
+        write!(f, r#""{}""#, project_field(next))?;
+        match sorted.peek() {
+            Some(vf) if Some(project_key(vf)) != key => close(f)?,
+            None => close(f)?,
+            Some(_) => f.write_str(", ")?,
+        }
```
Comment (jkomyno) [inline schema-engine/connectors/schema-connector/src/warnings.rs:50]: Thanks for this snippet of `Iterator::peek()` @tomhoule, now it's much clearer how it should be used.
It reminds me of a `std::next(iter)` wrapped in a `std::optional` in C++17

**Pull request merged**